src/hotspot/share/prims/jni.cpp

```diff
@@ -2876,7 +2876,7 @@ JNI_ENTRY(jweak, jni_NewWeakGlobalRef(JNIEnv *env, jobject ref))
   HOTSPOT_JNI_NEWWEAKGLOBALREF_ENTRY(env, ref);
   Handle ref_handle(thread, JNIHandles::resolve(ref));
   jweak ret = JNIHandles::make_weak_global(ref_handle, AllocFailStrategy::RETURN_NULL);
-  if (ret == nullptr) {
+  if (ret == nullptr && ref_handle != nullptr) {
```

Comment:
That isn't how you check a Handle for null content - you need `!ref_handle.is_null()`.

Reply:
even better would be `ref_handle.not_null()`.  Though I think the comparison does work.

Reply:
Yes it seems that some implicit Handle conversions were removed at some point which now makes direct comparison to `nullptr` allowable - somewhat obsolescing the `is_null()/not_null()` helper methods.